src/game_objects/construct.cpp

```diff
@@ -53,19 +55,27 @@ Spikes::Spikes(int x, int z, int id) : Construct(x, z, id) {
         rigid.is_ghost = true;
         rigid.position = { x, 0.0f, z };
         events::add_rigidbody_event(rigid);
+        notify_on_collision = true;
     }
 }
 
 void Spikes::s_on_collision(GameObject *other) {
     Player *player = dynamic_cast<Player*>(other);
     if (player) {
-        events::dungeon::network_interact_event(other->get_id(), id);
+        events::dungeon::network_collision_event(other->get_id(), id);
+        if (player->s_take_damage()) {
```

Comment:
Since we have no concept of player health, I think leaving this blank is fine.  I can see a use case for this if we want to adjust how much gold we drop based on what the collision was, but I think for now it's okay to leave it as-is.